src/Helper/Service/BasePathFactory.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\View\Helper\Service;
+
+use ArrayAccess;
+use Laminas\View\Helper\BasePath;
+use Psr\Container\ContainerInterface;
+
+use function assert;
+use function is_array;
+use function is_string;
+
+final class BasePathFactory
+{
+    public function __invoke(ContainerInterface $container): BasePath
+    {
+        $config = $container->has('config')
+            ? $container->get('config')
+            : [];
+
+        assert(is_array($config) || $config instanceof ArrayAccess);
+
+        // The expected location in config for the base path in an MVC application
+        // is config.view_manager.base_path
+        // @link https://docs.laminas.dev/laminas-mvc/services/#viewmanager
```

Comment:
And in a Mezzio application?

Reply:
I've been thinking about this, Mezzio and MVC have disparate config keys for the same things, particularly templates. the `view_manager` top level key, `view_helper_config`, `view_helpers` all have cross over I think. In an ideal world view would define a single top level key of `laminas_view` with everything underneath - maybe we can sort this out for 3.0 and then leave the Mezzio/MVC integration packages to sort out their own config structure. Anyhow, for the purposes of this pull, from searching around, I found `view_manager.base_path` to be the conventional default. Not convinced that base path is even useable by default in Mezzio as per recent Slacks…

Reply:
> …maybe we can sort this out for 3.0…

laminas-mvc-view is the way should go.